test_pull_requests

```diff
@@ -152,6 +178,97 @@ def submitted_tests_for_branch(branch)
   $submitted_tests[branch] ? $submitted_tests[branch] : $submitted_tests['*']
 end
 
+# Pulls out as much metadata from the bot comment as it can, including
+# state (AKA content), Jenkins build URL, base repo commit ID, and the
+# extended tests metadata
+def extract_bot_comment_fields(bot_comment_body, settings=nil)
+  test_prefix = nil
+  content = nil
+  build_url = nil
+  base_commit = nil
+  extended_tests = nil
+  state = nil
+  build_url_pattern = nil
+  if settings.nil?
+    # Use this test to set the build url matching pattern
+    if ENV['BUILD_URL']
+      # Don't set build_url directly from the env var, since we want
+      # to make sure they match
+      build_url_pattern = / \((#{ENV['BUILD_URL']})\)/
+    else
+      build_url_pattern = / \((https?:[^)]+)\)/
+    end
+    # Do our best to find the prefix since we don't know it beforehand
+    if bot_comment_body =~ /^([^:(]+): /
+      test_prefix = $1.strip
+    elsif bot_comment_body =~ /^([^ (]+) /
+      test_prefix = $1.strip
+    end
+  else
+    build_url_pattern = / \((#{settings['jenkins_url']}[^)]+)\)/
+    test_prefix = settings['test_prefix']
+  end
+  # Do our best to find the content (aka state) even if there aren't metadata in parens after it
+  if test_prefix && bot_comment_body =~ /^#{test_prefix} ([^(]+) ?\(?/
+    content = $1.strip
+    # Match up to colon or space since "Waiting" content varies
+    state = CONTENT_TO_STATE[content[/^(#{CONTENT_EVALUATING}|[^: ]+)/, 1]]
```

Comment:
An explanatory variable for `content[/^(#{CONTENT_EVALUATING}|[^: ]+)/, 1]` could be useful here